lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,52 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = { :kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776 }
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      raise(::Proxy::Error::ConfigurationError, 'Unable to continue - must provide a path.') if opts[:path].nil?
+      @pathes_hash = path_hash(opts[:path])
+      @path = @pathes_hash.values.join(' ')
+      @size = SIZE[opts[:size]] || SIZE[:gigabyte]
+      @stat = {}
+      find_df
+      get_stat
+    end
+
+    def to_json
+      stat.to_json
+    end
+
+    private
+
+    attr_reader :command_path
+
+    def find_df
+      @command_path = which('df') || raise(::Proxy::Error::ConfigurationError, 'df command was not found unable to retrieve usage information.')
+    end
+
+    def command
+      "LANG=C #{command_path} -B#{size} #{path}"
+    end
+
+    def get_stat
+      raw = Open3::popen3(command) do |stdin, stdout, stderr, thread|
+        stdout.read.split("\n")
+      end
+
+      replacers = {"mounted on" => :mounted, "use%" => :percent}
+      titles = raw.shift.downcase.gsub(/(use%|mounted on)/) { |m| replacers.fetch(m,m)}.split.map(&:to_sym)
```

Comment:
Also GNU df supports column selectors via an option, since we already require GNU df apparently (BSD only supports 1k blocks), we could use that and parsing could be more simple.